mkosi/__init__.py

```diff
@@ -3645,46 +3645,56 @@ def make_generated_root(args: CommandLineArguments, root: Path, for_cache: bool)
     return None
 
 
-def read_partition_table(loopdev: Path) -> Tuple[List[str], int]:
-    table = []
-    last_sector = 0
+@dataclasses.dataclass
+class PartitionTable:
+    partitions: List[str]
+    last_partition_sector: int
 
-    c = run(["sfdisk", "--dump", loopdev],
-            stdout=PIPE,
-            universal_newlines=True)
+    @classmethod
+    def read(cls, loopdev: Path) -> 'PartitionTable':
```

Comment:
Add `from __future__ import annotations` at the very top of the file and you can remove the string quotation marks from `PartitionTable` here

Reply:
Thanks, I didn't know this is possible. I did that, and also pushed one more commit to use the same trick in other places.